src/components/SignUpPage.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { addUser, register } from "./firebaseConnect";
+
+const SignupPage = () => {
+    const [user, setUser] = useState({
+        name: "",
+        email: "",
+        password: "",
+        number: "",
+        address: ""
+    });
+    const navigate = useNavigate();
+
+    const [isLoading, setIsLoading] = useState(false);
+
+    const createUser = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
We're not creating user here instead storing user input only. So, the function can be like `handleUserInput`, something like this, @swornex